test/network.spec.js

```diff
@@ -140,6 +140,43 @@ describe('Request.postData', function() {
   });
 });
 
+describe('Request.postDataJSON', function () {
+  it('should work with application/json content-type', async ({ page, server }) => {
+    await page.goto(server.EMPTY_PAGE);
+    server.setRoute('/post', (req, res) => res.end());
+    let request = null;
+    page.on('request', r => request = r);
+    await page.evaluate(() => fetch('./post', { method: 'POST', body: JSON.stringify({ foo: 'bar' }), headers: new Headers({ 'Content-Type': 'application/json' }) }));
+    expect(request).toBeTruthy();
+    expect(request.postDataJSON()).toEqual({ "foo": "bar" });
+  });
+  it('should be |undefined| when it is has no application/json content-type', async ({ page, server }) => {
+    await page.goto(server.EMPTY_PAGE);
+    server.setRoute('/post', (req, res) => res.end());
+    let request = null;
+    page.on('request', r => request = r);
+    await page.evaluate(() => fetch('./post', { method: 'POST', body: JSON.stringify({ foo: 'bar' }) }));
+    expect(request).toBeTruthy();
+    expect(request.postDataJSON()).toEqual(null);
+  });
+
+  it("should parse the data if content-type is application/x-www-form-urlencoded", async({page, server}) => {
+    await page.goto(server.EMPTY_PAGE);
+    server.setRoute('/post', (req, res) => res.end());
+    let request = null;
+    page.on('request', r => request = r);
+    await page.setContent(`<form method='POST' action='/post'><input type='text' name='foo' value='bar'><input type='number' name='baz' value='123'><input type='submit'></form>`);
+    await page.click('input[type=submit]');
+    expect(request).toBeTruthy();
+    expect(request.postDataJSON()).toEqual({"foo":"bar","baz":"123"});
```

Comment:
nit: we use single quotes.

Reply:
Would be cool to share a style guide somewhere, Visual Studio Code e.g. adds spaces before curly brackets etc. but thats not common in this project.